source/MaterialXContrib/MaterialXMaya/Plugin.cpp

```diff
@@ -19,73 +19,122 @@
+    {
+        MStringArray mstrArray;
+        if (MGlobal::executeCommand("optionVar -q " + optionVarName, mstrArray))
+        {
+            for (const MString& mstrPath : mstrArray)
+            {
+                append(mx::FilePath(mstrPath.asChar()));
+            }
+        }
+    }
+
+  private:
+    mx::FileSearchPath& _searchPath;
+    std::unordered_set<std::string> _uniquePaths;
+};
+
+void setIntermediateDumpPath()
 {
-    // Always include plug-in load path
-    _pluginLoadPath = pluginLoadPath;
+    bool optionVarExists = false;
+    MString path = MGlobal::optionVarStringValue("materialXIntermediateDumpPath", &optionVarExists);
 
-    // Search in standard installed resources directories and plug-in relative resources
-    _resourceSearchPath.append(_pluginLoadPath);
-    _resourceSearchPath.append(_pluginLoadPath / mx::FilePath("../../resources"));
-    _resourceSearchPath.append(_pluginLoadPath / mx::FilePath("../resources"));
+    if ( !optionVarExists || path.length() == 0 )
+    {
+        return;
+    }
 
     MHWRender::MRenderer* const theRenderer = MHWRender::MRenderer::theRenderer();
-    MHWRender::MFragmentManager* const fragmentManager = theRenderer ? theRenderer->getFragmentManager() : nullptr;
+    if (!theRenderer)
+    {
+        MGlobal::displayError("Failed to get the VP2 renderer");
+        return;
+    }
+
+    MHWRender::MFragmentManager* const fragmentManager = theRenderer->getFragmentManager();
     if (!fragmentManager)
     {
         MGlobal::displayError("Failed to get the VP2 fragment manager");
+        return;
     }
-    else
-    {
-        // Set to resource path for now
-        std::string shaderDebugPath = _resourceSearchPath[1].asString();
 
-        // Add explicitly as VP2 does no prepend a folder separator, and thus fails silently to output anything.
-        shaderDebugPath += "/";
-        fragmentManager->setEffectOutputDirectory(shaderDebugPath.c_str());
-        fragmentManager->setIntermediateGraphOutputDirectory(shaderDebugPath.c_str());
+    switch (path.asChar()[path.length() - 1])
+    {
+    case '/':
+    case '\\':
+        break;
+
+    default:
+        // Add explicitly as VP2 does not append a folder separator,
+        // and thus fails silently to output anything.
+        path += "/";
     }
 
+    fragmentManager->setEffectOutputDirectory(path);
+    fragmentManager->setIntermediateGraphOutputDirectory(path);
+}
+
+} // anonymous namespace
+
+Plugin& Plugin::instance()
+{
+    static Plugin s_instance;
+    return s_instance;
+}
+
+void Plugin::initialize(const std::string& pluginLoadPath)
+{
+    // Always include plug-in load path
+    _pluginLoadPath = pluginLoadPath;
+    setIntermediateDumpPath();
     loadLibraries();
 }
 
-void Plugin::loadLibraries()
+mx::FileSearchPath Plugin::getResourceSearchPath() const
 {
-    _librarySearchPath = mx::FileSearchPath();
-    _libraryDocument = mx::createDocument();
+    mx::FileSearchPath searchPath;
```

Comment:
Not caching the resource path in the plugin instance anymore, instead generating it based on the optionVar every time it's queried.